chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/measurefix/MeasureFixJob.java

```diff
@@ -0,0 +1,132 @@
+        LOGGER.info("********* Starting the G1/G2 Measure Data Fix job. *********");
+        try {
+            setSecurityContext();
+            ExecutorService executorService = Executors.newFixedThreadPool(threadCount);
+
+            List<CompletableFuture<Boolean>> futures = getAll2015CertifiedProducts().stream()
+                    .map(cp -> CompletableFuture.supplyAsync(() ->
+                            resaveListing(cp), executorService))
+                    .collect(Collectors.toList());
+
+            CompletableFuture<Void> combinedFutures = CompletableFuture
+                    .allOf(futures.toArray(new CompletableFuture[futures.size()]));
+
+            // This is not blocking - presumably because the job executes using it's own ExecutorService
+            // This is necessary so that the system can indicate that the job and it's threads are still running
+            combinedFutures.get();
+            LOGGER.info("********* Completed the G1/G2 Measure Data Fix job. *********");
+        } catch (Exception e) {
+            LOGGER.catching(e);
+        }
+    }
+
+    private List<CertifiedProductDetailsDTO> getAll2015CertifiedProducts() {
+        LOGGER.info("Retrieving all 2015 listings");
+        List<CertifiedProductDetailsDTO> listings = certifiedProductDAO.findByEdition(
+                CertificationEditionConcept.CERTIFICATION_EDITION_2015.getYear());
+        LOGGER.info("Completed retreiving all 2015 listings");
+        return listings;
+    }
+
+    private Boolean resaveListing(CertifiedProductDetailsDTO cp) {
+        // We need to manually create a transaction in this case because of how AOP works. When a method is
+        // annotated with @Transactional, the transaction wrapper is only added if the object's proxy is called.
+        // The object's proxy is not called when the method is called from within this class. The object's proxy
+        // is called when the method is public and is called from a different object.
+        // https://stackoverflow.com/questions/3037006/starting-new-transaction-in-spring-bean
+        TransactionTemplate txTemplate = new TransactionTemplate(txManager);
+        txTemplate.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
+        return txTemplate.execute(new TransactionCallback<Boolean>() {
+            @Override
+            public Boolean doInTransaction(TransactionStatus status) {
+                try {
+                    CertifiedProductSearchDetails listing = certifiedProductDetailsManager.getCertifiedProductDetails(cp.getId());
+                    ListingUpdateRequest request = ListingUpdateRequest.builder()
+                            .listing(listing)
+                            .acknowledgeWarnings(true)
+                            .reason(null)
+                            .build();
+
+                    certifiedProductManager.update(request, false);
```

Comment:
I know we agreed not to log activity so we didn't have all those activities as a side-effect of this job. I'm also now thinking that this will change the lastUpdated date for each of these listings (even ones that are valid and won't have their g1/g2 measures changed at all) and I just want to make sure that we don't mind having that happen. Something to ask at the feature review maybe but I thought of it now and wanted to write it down.